lib/msprime.c

```diff
@@ -2238,10 +2252,18 @@ msp_dtwf_generation(msp_t *self)
         avl_init_tree(&Q[i], cmp_segment_queue, NULL);
     }
 
+    num_links = fenwick_get_total(&self->links);
+    ret = msp_sanity_check(self, num_links);
+    if (ret != 0) {
+        goto out;
+    }
+    msp_verify_segments(self);
```

Comment:
This will kill performance, so we'll need to remember to remove this before merging.

Reply:
Also, might as well use msp_verify() if you're doing this, as this should check some other stuff too. (this is exactly the right thing to do while developing the algorithms though!)